src/cluster_legacy.c

```diff
@@ -6540,20 +6540,20 @@ unsigned int delKeysInSlot(unsigned int hashslot) {
     for (int i = 0; i < server.dbnum; i++) {
         kvstoreHashtableIterator *kvs_di = NULL;
         void *next;
-        serverDb db = server.db[i];
-        kvs_di = kvstoreGetHashtableIterator(db.keys, hashslot, HASHTABLE_ITER_SAFE);
+        serverDb *db = server.db[0];
```

Comment:
Shouldn't this be `i`?

And should we skip uninitialized databases?

```suggestion
        serverDb *db = server.db[i];
        if (db == NULL) continue;
```

Reply:
Yes, we should. Bad bug. Fixing.